src/app/shared/components/phone-number-formater/phoneFormat.ts

```diff
@@ -0,0 +1,46 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[appPhoneFormat]'
+})
+export class PhoneFormatDirective {
+
+  constructor(private el: ElementRef) {}
+
+  @HostListener('input', ['$event']) 
+  @HostListener('keydown', ['$event'])
+  onInput(event: any) {
```

Comment:
This works for most cases. Another option is to subscribe to changes in the form value, so the formatter fires even if someone pastes a number into the field (no `keydown` event). The downside of that is if the control value is ever invalid then you have to take an additional step to set it to something valid again (no event to stop propagation on).

Reply:
I wouldn't worry about changing anything here for this PR though

Reply:
Thanks, I will take that into consideration next time!